packages/fuchsia_remote_debug_protocol/lib/src/dart/dart_vm.dart

```diff
@@ -122,9 +122,10 @@ class DartVm {
   Future<List<IsolateRef>> getMainIsolatesByPattern(Pattern pattern) async {
     final Map<String, dynamic> jsonVmRef =
         await invokeRpc('getVM', timeout: _kRpcTimeout);
-    final List<Map<String, dynamic>> jsonIsolates = jsonVmRef['isolates'];
     final List<IsolateRef> result = <IsolateRef>[];
-    for (Map<String, dynamic> jsonIsolate in jsonIsolates) {
+    // Grabs key within for-loop to prevent possible casting issues, as this key
+    // can return a `List<dynamic>` instead of `List<Map<String, dynamic>>`.
```

Comment:
"can" return? why wouldn't it always return that?

Reply:
Frankly I'm not sure.  Different versions of the json_rpc_2 library?  I ran into this issue when running the driver on a Fuchsia device (whereas it had never come up locally).